components/clp-py-utils/clp_py_utils/s3_utils.py

```
@@ -304,5 +308,44 @@ def s3_put(s3_config: S3Config, src_file: Path, dest_file_name: str) -> None:
+    `s3_config.key_prefix`.
+    Note: `boto3.delete_objects` is limited to 1000 objects per call. This method divides the input
+    object keys into chunks of 1000 and calls `boto3.delete_objects` multiple times if necessary.
+
+    :param s3_config: S3 configuration specifying the upload destination and credentials.
+    :param object_keys: The set of object keys to delete, relative to `s3_config.key_prefix`.
+    :raises: Propagates `boto3.client`'s exceptions.
+    :raises: Propagates `boto3.client.delete_object`'s exceptions.
+    """
+
+    def chunked_iterate(key_prefix: str, relative_object_keys: Set[str], chunk_size: int):
+        chunk: List[str] = list()
```

Pylint (3.3.7)

[refactor] Consider using [] instead of list()

(R1734)